src/phase0/state_transition/epoch_processing.rs

```diff
@@ -177,8 +178,29 @@ pub fn process_slashings<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    let epoch = get_current_epoch(state, context);
+    let total_balance = get_total_active_balance(state, context)?;
+    let adjusted_total_slashing_balance = Gwei::min(
+        state.slashings.iter().sum::<Gwei>() * context.proportional_slashing_multiplier,
```

Comment:
Yeah idk why I needed that, but if I remove it, I get an error:

> cannot satisfy `_: std::iter::Sum<&u64>`
help: consider specifying the type argument in the method call

Reply:
i think we can just get away with `state.slashings.iter().copied().sum()`

the error suggests it doesn't like the `&Gwei` from the plain `iter` and i think `into_iter` will fail bc it would try to consume the leading ref

usually you could reach for `cloned` in this situation but bc we know our type is `Copy` we will use `copied` (see answers for reasoning: https://stackoverflow.com/questions/58413583/difference-between-copied-and-cloned-on-rust-iterators)

Reply:
Hmm so none of those solved it either -- what [I'm reading](https://github.com/rust-lang/rust/issues/25094) is that `sum()` is super picky with type inferences and needs to have a type specified. For example, when I remove the multiplication by `context.proportional_slashing_multiplier`, it doesn't complain & compiles:
```
let adjusted_total_slashing_balance = Gwei::min(state.slashings.iter().sum(), total_balance);
```
...which is sort of confusing as well since `total_balance` is the same type as `context.proportional_slashing_multiplier`, so it seems like it gets super picky once operations are attempted.

Reply:
hmm interesting, well thanks for looking into it!